cmd/setup_challenges.go

```diff
@@ -111,6 +111,19 @@ func setupDNS(ctx *cli.Context, client *lego.Client) {
 		log.Fatal(err)
 	}
 
+	switch {
+	case ctx.IsSet("ipv4only") && ctx.IsSet("ipv6only"):
+		// If both flags are set then it's as good as not providing either flag,
+		// so we default to the OS choice.
+		dns01.SetNetworkStack("both")
+
+	case ctx.IsSet("ipv4only"):
+		dns01.SetNetworkStack("ipv4")
+
+	case ctx.IsSet("ipv6only"):
+		dns01.SetNetworkStack("ipv6")
```

Comment:
... and change this to:

```suggestion
	case ctx.IsSet("ipv4only") && ctx.IsSet("ipv6only"):
		// If both flags are set then it's as good as not providing either flag,
		// so we default to the OS choice.
		dns01.SetNetworkStack(dns01.DefaultNetworkStack)

	case ctx.IsSet("ipv4only"):
		dns01.SetNetworkStack(dns01.IPv4Only)

	case ctx.IsSet("ipv6only"):
		dns01.SetNetworkStack(dns01.IPv6Only)
```